novawallet/Common/ViewModel/DisplayAddressViewModelFactory.swift

```diff
@@ -96,4 +101,42 @@ final class DisplayAddressViewModelFactory: DisplayAddressViewModelFactoryProtoc
             imageViewModel: poolIcon
         )
     }
+
+    func createViewModel(from model: DisplayAddress) -> DisplayAddressViewModel {
+        createViewModel(from: model, chainFormat: nil)
+    }
+
+    func createViewModel(from model: DisplayAddress, using chainFormat: ChainFormat) -> DisplayAddressViewModel {
+        createViewModel(from: model, chainFormat: chainFormat)
+    }
+
+    func createViewModel(
+        from formattedCallAccount: FormattedCall.Account,
+        chain: ChainModel
+    ) throws -> DisplayAddressViewModel {
+        let accountAddress = try formattedCallAccount.accountId.toAddress(using: chain.chainFormat)
+
+        let name: String?
+        let imageViewModel: ImageViewModelProtocol?
+
+        switch formattedCallAccount {
+        case let .local(localAccount):
+            imageViewModel = accountIconFactory.createDrawableIconViewModel(
+                from: localAccount.walletIdenticonData
+            )
+            name = localAccount.chainAccount.name
+        case let .remote(accountId):
+            imageViewModel = accountIconFactory.createIdentifiableDrawableIconViewModel(
```

Comment:
For addresses better to display the icon based on `PolkadotIconGenerator`. Also make sure we truncate addresses in the middle